src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/TrackingOptInModal.java

```diff
@@ -20,172 +20,180 @@
+  }
+
+  public void rejectOptInModal(WikiaWebDriver driver, String country, Page page,
+                               String[] instantGlobals) {
+    setGeoCookie(driver, EU_CONTINENT, country);
+    getUrl(urlWithInstantGlobals(instantGlobals, page));
+    clickRejectButton();
+  }
+
+  public void verifyTrackingRequestsNotSend(List<String> urlPatterns,
+                                            NetworkTrafficInterceptor networkTrafficInterceptor) {
+    isTrackingRequestsNotSend(urlPatterns, networkTrafficInterceptor);
+  }
+
+  public void verifyTrackingRequestsSend(List<String> urlPatterns,
+                                         NetworkTrafficInterceptor networkTrafficInterceptor) {
+    isTrackingRequestSend(urlPatterns, networkTrafficInterceptor);
+  }
+
+  public String urlOptInModalDisplayedOasis(Page page) {
+    return urlBuilder.appendQueryStringToURL(urlBuilder.getUrlForPage(page), MODAL_INSTANT_GLOBAL);
+  }
+
+  public void isTrackingRequestsNotSend(List<String> elementsList,
+                                        NetworkTrafficInterceptor networkTrafficInterceptor) {
+    wait.forX(WAITNG_TIME_FOR_ALL_REQUESTS);
+    for (int i = 0; i < elementsList.size(); i++) {
+      Assertion.assertFalse(
+          isSuccessfulResponseByUrlPattern(networkTrafficInterceptor, elementsList.get(i)),
+          "Request to " + elementsList.get(i) + " services was found");
+    }
+  }
+
+  public void isTrackingRequestSend(List<String> elementsList,
+                                    NetworkTrafficInterceptor networkTrafficInterceptor) {
+    for (int i = 0; i < elementsList.size(); i++) {
+      wait.forSuccessfulResponseByUrlPattern(networkTrafficInterceptor, elementsList.get(i));
+    }
+  }
+
+  public boolean areResponsesByUrlPatternSuccessful(List<String> elementsList,
+                                                    NetworkTrafficInterceptor networkTrafficInterceptor) {
+    boolean result = true;
+    for (int i = 0; i < elementsList.size(); i++) {
+      try {
+        wait.forSuccessfulResponseByUrlPattern(networkTrafficInterceptor, elementsList.get(i));
+      } catch (Exception e) {
+        PageObjectLogging
+            .log("Did not get successfull response with element: " + elementsList.get(i),
+                 e, false);
+        result = false;
+      }
+    }
+    return result;
+  }
+
+  private boolean isSuccessfulResponseByUrlPattern(
+      final NetworkTrafficInterceptor trafficInterceptor,
+      final String pattern) {
+    try {
+      entry = trafficInterceptor.getEntryByUrlPattern(pattern);
+      return entry.getResponse().getStatus() < 400;
+    } catch (NullPointerException ex) {
+      return false;
+    }
+  }
+
+  private static String appendTrackingOptOutParameters(String url, String[] instantGlobals) {
+    UrlBuilder urlBuilder = new UrlBuilder();
+
+    for (String instantGlobal : instantGlobals) {
+      url = urlBuilder.globallyEnableGeoInstantGlobalOnPage(url, instantGlobal);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Introduce a new variable instead of reusing the parameter "url". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AS1226)